window_manager/src/main.cpp

```diff
@@ -0,0 +1,97 @@
+#include <rcbe-engine/core/WindowManager.hpp>
+#include <rcbe-engine/utils/output_utils.hpp>
+#include <rcbe-engine/datamodel/system/window_config.hpp>
+#include <rcbe-engine/core/AbstractInputManager.hpp>
+#include <rcbe-engine/core/EditorInputManager.hpp>
+#include <rcbe-engine/datamodel/visual/Texture.hpp>
+
+#include <boost/log/trivial.hpp>
+
+int main(int argc, char* argv[]) {
+    try {
+        rcbe::utils::setup_logging();
+        rcbe::core::WindowManager wm { true };
+
+        rcbe::core::window_config window_conf;
+        window_conf.caption = "Testing TGA";
+        window_conf.position = rcbe::math::Vector2d { 0.0, 0.0 };
+        window_conf.background_color = rcbe::visual::RGBAColor { 0.0, 0.0, 0.0, 1.0 };
+        window_conf.editor = true;
+        window_conf.fullscreen = false;
+        window_conf.input_scheme = "";
+        window_conf.size = rcbe::core::Dimensions { 1240, 720 };
+        window_conf.process_input = true;
+        window_conf.type = decltype(window_conf.type)::drawing_window;
+
+        auto window = wm.createWindow(std::move(window_conf));
+
+        rcbe::core::EditorInputManager::HandlerCollection handlers = {
+                {
+                    rcbe::core::InputEventType::button_press, [window](
+                        rcbe::core::InputManagerImplementation& im,
+                        rcbe::core::InputManagerImplementation::InputEventReference event,
+                        rcbe::core::InputManagerImplementation::PreviousEventReference previous
+                    ) {
+                        int x=event.xbutton.x, y=event.xbutton.y;
+                        BOOST_LOG_TRIVIAL(debug) << "Mouse button pressed " << x << " " << y;
+
+                        auto optional_gc = window->getGraphicContext();
+
+                        const auto red_pixel = 0xFF0000; // for some strange reason if specified with alpha channel (even zeroed out)
```

Comment:
Maybe if reverse in `operator uint32_t` is removed this will work out?

Reply:
Can't fix. Added TODO.